algorithms/decision_engines/mlp.py

```diff
@@ -229,10 +228,18 @@ def _calculate(self, syscall: Syscall):
             returns: anomaly score
         """
         input_vector = self.input_vector.get_result(syscall)
-        label = self.output_label.get_result(syscall)
```

Comment:
This must be executed even if input_vector is None. -> the underlying computation DAG must be called otherwise this could result in unexpected behavior.